File: back/app/core/roles.py
"""Role names.

Kept in `core` so nothing has to import ORM models just to name a role.
"""

ROLE_ADMIN = "admin"
ROLE_STAFF = "staff"
ROLE_MEMBER = "member"

DEFAULT_ROLES = (ROLE_ADMIN, ROLE_STAFF, ROLE_MEMBER)

File: front/lib/auth/roles.ts
import type { Role } from "@/lib/auth/types"

// Mirrors back/app/core/roles.py — keep in sync.
export const ROLE_ADMIN = "admin"
export const ROLE_STAFF = "staff"
export const ROLE_MEMBER = "member"

export function hasRole(user: { roles: Role[] }, ...names: string[]): boolean {
  return user.roles.some((r) => names.includes(r.name))
}

export function isAdmin(user: { roles: Role[] }): boolean {
  return hasRole(user, ROLE_ADMIN)
}
